### ISSUE TITLE
ATEM causing flicker in tally color

### ISSUE THREAD

**Issue by jerriley** [josephdadams/TallyArbiter#67]
I have the ATEM (2 M/E Production 4K; firmware version 8.5.3) set up properly with Tally Arbiter and M5StickC+ and M5 ATOM Matrix. One minor bug that I noticed is that the color flickers slightly on every input change of the ATEM. This happens on both the M5StickC+ and the M5 ATOM Matrix. I don't notice it on Tally Arbiter settings but the flicker is so quick that perhaps it is there and I'm just missing it. There is no flicker when using OBS instead of ATEM. See video:

https://user-images.githubusercontent.com/35228159/110219117-2a8a2480-7e72-11eb-9c36-d745b271372a.mov

**Comment by josephdadams** [comment]
It looks yellow before red. Yellow is the color used for when it’s in preview and program at the same time. Are you doing a t bar or auto transition in this clip instead of a cut? If that’s the case then I’m pretty sure that’s what is happening.

**Comment by jerriley** [comment]
This is only with cutting. I've tried cutting using space bar on laptop as well as Cut button on hardware panel (1 M/E Advanced Panel). I have CUT_BUS = true, so I don't expect to see any yellow color. I'm not sure if I see a clear frame of video which shows yellow color. I'm examining the logs more closely now. It seems the tally information is sent many times for each cut on the ATEM. Looking at the Arduino Serial Monitor, after I press Cut one time, the monitor says "The device is in preview" several times. And in the Tally Arbiter settings under Tally Data, after one Cut press, there is one message for each Device. Does that make sense? I think the color is updating once per Device for each Cut. And when the color is red and it updates to red, it manifests as a flicker as shown in the video.

**Comment by jerriley** [comment]
Using ATEM. If I have 6 Devices (cameras) set up. Then I preview one camera. I get this in Arduino Serial Monitor:

`13:55:18.016 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.051 -> The device is in preview.
13:55:18.086 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.086 -> The device is in preview.
13:55:18.119 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.152 -> The device is in preview.
13:55:18.186 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.220 -> The device is in preview.
13:55:18.254 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.291 -> The device is in preview.
13:55:18.324 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.324 -> The device is in preview.
13:55:18.357 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.391 -> The device is in preview.
13:55:18.424 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.462 -> The device is in preview.
13:55:18.496 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.496 -> The device is in preview.
13:55:18.563 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.563 -> The device is in preview.
13:55:18.596 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.630 -> The device is in preview.
13:55:18.664 -> Got event 'device_states', data: [{"deviceId":"8fa26f6e","busId":"e393251c","sources":[{"sourceId":"2acbb58f","address":"2"}],"linkedSources":[],"active":true},{"deviceId":"8fa26f6e","busId":"334e4eda","sources":[],"linkedSources":[]}]
13:55:18.699 -> The device is in preview.
`

If I have only one Device (camera) set up, then I only get "The device is in preview" twice. Still flicker.

In OBS, even with many Devices I still only get "The device is in preview" once (although I also get "The device is in preview+program" which is a separate bug).

**Comment by jerriley** [comment]
Thanks for checking. I installed the latest version of Tally Arbiter (Electron app), and this flickering issue persists. I flashed the M5 Atom with the latest code from themusicnerd (fixes a couple things), and this flickering issue persists.

The flicker happens when cutting and when transitioning (mix). The flicker also happens even if the tally color doesn't change, but the preview or program bus changes. For example, if tally light 1 is red (program), and I change the preview (green) from 2 to 3, then tally light 1 will flicker red while staying red.
